e2e/test/iothub/messaging/AzureSecurityCenterForIoTSecurityMessageE2ETests.cs

```diff
@@ -174,23 +174,17 @@ private async Task TestSecurityMessageModuleAsync(Client.TransportType transport
             AzureSecurityCenterForIoTLogAnalyticsClient logAnalticsTestClient)
         {
             await moduleClient.OpenAsync().ConfigureAwait(false);
-
-            Client.Message testMessage = ComposeD2CSecurityTestMessage(out string eventId, out string payload, out string p1Value);
+            Client.Message testMessage = ComposeD2CSecurityTestMessage(out string eventId, out _, out _);
             await moduleClient.SendEventAsync(testMessage).ConfigureAwait(false);
 
-            await ValidateEventAsync(deviceId, testMessage, eventId, payload, p1Value, logAnalticsTestClient).ConfigureAwait(false);
+            await ValidateEventAsync(deviceId, eventId, logAnalticsTestClient).ConfigureAwait(false);
         }
 
         private async Task ValidateEventAsync(
             string deviceId,
-            Client.Message message,
             string eventId,
-            string payload,
-            string p1Value,
             AzureSecurityCenterForIoTLogAnalyticsClient logAnalticsTestClient)
         {
-            bool isReceivedEventHub = EventHubTestListener.VerifyIfMessageIsReceived(deviceId, message, payload, p1Value, TimeSpan.FromSeconds(10));
-            Assert.IsFalse(isReceivedEventHub, "Security message received in customer event hub.");
```

Comment:
For this pattern of verification, is there a place where we are still checking for ack from iothub?

Reply:
Not explicitly, no. If the `SemdEventAsync()` call completes successfully, then the call was implicitly ack'ed by the service. If the response is not a success, then either the application retries (based on the retry policy), or forwards the exception to the calling application.